### DISCUSSION TITLE
Pynetbox get devices by cluster

### DISCUSSION THREAD

**Discussion by shanmugara** [netbox-community/netbox#9087] [Q&A] [answered]
I am trying get all `devices` that belong to a specific `cluster`, been playing around with nb.virtualization, but not able to find a way to do that. Is is possible to get devices by cluster? Thank you.

**Comment by candlerb** [comment, marked as answer]
```
>>> nb.virtualization.clusters.get(name="nuc-ganeti").id
3
>>> list(nb.dcim.devices.filter(cluster_id=3))
[nuc1, nuc2]
```

The REST API filtering is less powerful than the Django filtering, e.g. you can't do `cluster__name="nuc-ganeti"` (and such unimplemented query parameters are silently ignored)

**Reply by shanmugara** [reply]
That works perfect! Thank you.
